airbyte-server/src/main/java/io/airbyte/server/config/ApplicationBeanFactory.java

```diff
@@ -83,6 +83,14 @@ public Path workspaceRoot(@Value("${airbyte.workspace.root}") final String works
     return Path.of(workspaceRoot);
   }
 
+  @Singleton
+  @Requires(env = WorkerMode.CONTROL_PLANE)
+  public JsonSecretsProcessor jsonSecretsProcessor(final FeatureFlags featureFlags) {
```

Comment:
@gosusnp I believe that this was added because originally the code loaded when in `data-plane` mode did not require this singleton.  I'm not sure if that is still the case.  If that has changed and it is now required, we should remove the `@Requires` annotation.  If it is not used by anything in `data-plane` mode, we should leave this to avoid instantiating a singleton that we don't need.  cc: @benmoriceau